distributed/tests/test_scheduler.py

```diff
@@ -615,11 +615,19 @@ async def test_ready_remove_worker(s, a, b):
 
 @gen_cluster(client=True, Worker=Nanny, timeout=60)
 async def test_restart(c, s, a, b):
+    from distributed.scheduler import TaskState
+
+    before = TaskState._instances
```

Comment:
You're creating a new reference to the same WeakSet. The assertion below is always true because it's the same object

Reply:
you're right of course. I don't even know if this is a sane test, to be honest. We've been looking into TaskState garbage collection before and this is a bit tricky. I'll probably just remove this